File: backend/python-api/utils/logger.py
import logging
import os
from datetime import datetime

# 로깅 설정
def setup_logger(name, log_file=None, level=logging.INFO):
    """로거 설정"""
    logger = logging.getLogger(name)
    logger.setLevel(level)
    
    # 이미 핸들러가 있으면 추가하지 않음
    if logger.handlers:
        return logger
    
    formatter = logging.Formatter(
        '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
    )
    
    # 콘솔 핸들러
    console_handler = logging.StreamHandler()
    console_handler.setFormatter(formatter)
    logger.addHandler(console_handler)
    
    # 파일 핸들러 (선택사항)
    if log_file:
        file_handler = logging.FileHandler(log_file)
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
    
    return logger

# 기본 로거
logger = setup_logger('zoop')

def info(message):
    """정보 로그"""
    logger.info(message)

def warn(message):
    """경고 로그"""
    logger.warning(message)

def error(message, exc_info=None):
    """에러 로그"""
    logger.error(message, exc_info=exc_info)

def debug(message):
    """디버그 로그"""
    logger.debug(message) 

File: frontend/src/utils/logger.js
// 로깅 유틸리티
const isDevelopment = process.env.NODE_ENV === 'development';

class Logger {
  static info(message, ...args) {
    if (isDevelopment) {
      console.log(`[INFO] ${message}`, ...args);
    }
  }

  static warn(message, ...args) {
    if (isDevelopment) {
      console.warn(`[WARN] ${message}`, ...args);
    }
  }

  static error(message, ...args) {
    // 에러는 항상 로깅
    console.error(`[ERROR] ${message}`, ...args);
  }

  static debug(message, ...args) {
    if (isDevelopment) {
      console.log(`[DEBUG] ${message}`, ...args);
    }
  }
}

export default Logger; 